doc-examples/hello-world-kotlin/build.gradle

```diff
@@ -0,0 +1,109 @@
+plugins {
+    id "org.jetbrains.kotlin.jvm" version "1.5.31"
+    id "org.jetbrains.kotlin.kapt" version "1.5.31"
+    id "org.jetbrains.kotlin.plugin.allopen" version "1.5.31"
+    id "io.micronaut.application" version "2.0.6"
+    id "com.google.protobuf" version "0.8.17"
+}
+
+ext {
+    grpcKotlinVersion = "1.2.0"
+}
+
+repositories {
+    mavenCentral()
+}
+
+mainClassName = "helloworld.Application"
+micronaut {
+    version micronautVersion
+    testRuntime "junit5"
+    enableNativeImage false
+}
+
+dependencies {
+    kapt("io.micronaut:micronaut-inject-java")
+    kapt("io.micronaut:micronaut-validation")
+
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:${kotlinVersion}")
+    implementation("org.jetbrains.kotlin:kotlin-reflect:${kotlinVersion}")
+    implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:${kotlinxCoroutinesVersion}")
+    implementation("io.micronaut:micronaut-runtime")
+    implementation("io.micronaut.grpc:micronaut-grpc-server-runtime")
+    implementation("io.micronaut.grpc:micronaut-grpc-client-runtime")
+
+    // tag::dependencies[]
+    implementation("io.grpc:grpc-kotlin-stub:${grpcKotlinVersion}")
+    compileOnly("io.grpc:grpc-stub:$grpcVersion")
+    // end::dependencies[]
+
+    runtimeOnly("ch.qos.logback:logback-classic")
+    runtimeOnly("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.0")
+
+    kaptTest("io.micronaut:micronaut-inject-java")
+
+    testImplementation("org.junit.jupiter:junit-jupiter-api")
+    testImplementation("io.micronaut.test:micronaut-test-junit5")
+    testImplementation("org.mockito:mockito-junit-jupiter")
+
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
+    testRuntimeOnly("org.jetbrains.spek:spek-junit-platform-engine:1.1.5")
+}
+
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile) {
+    kotlinOptions {
+        jvmTarget = "1.8"
+        javaParameters = true
+    }
+}
+
+if (JavaVersion.current() >= JavaVersion.VERSION_16) {
+    kotlin {
+        kotlinDaemonJvmArgs = [
+                "--illegal-access=permit",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+                "--add-opens=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED"
+        ]
+    }
+}
+
```

Comment:
I'm not particularly happy about these hacks. Kotlin 1.6.0 will fix kapt for JDK17 and then we would be able to upgrade everything.

Reply:
I agree, we can wait for the release, but it may take another month. So far, this is the only way to make Kotlin work for Java 16/17